api/api_test.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/ausocean/openfish/api/api"
+	"github.com/ausocean/openfish/api/ds_client"
+	"github.com/ausocean/openfish/api/handlers"
+	"github.com/gofiber/fiber/v2"
+)
+
+func TestCreateCaptureSource(t *testing.T) {
+	// Test Setup.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	// Send test request.
+	body := strings.NewReader(`{ "name": "Camera 1", "location": "-37.12345678,140.12345678" }`)
+	req := httptest.NewRequest("POST", "/api/v1/capturesources", body)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := app.Test(req, -1)
+	if err != nil {
+		t.Log(err.Error())
+		t.Fail()
+		return
+	}
+
+	// Check response.
+	if resp.StatusCode != fiber.StatusOK {
+		t.Log("expected HTTP code 200, got", resp.StatusCode)
+		t.Fail()
+	}
+}
+
+func TestGetCaptureSources(t *testing.T) {
+	// Test Setup.
+	ds_client.Init(false)
+	app := fiber.New()
+	RegisterAPIRoutes(app)
+
+	// Populate datastore.
+	body := strings.NewReader(`{ "name": "Camera 1", "location": "-37.12345678,140.12345678" }`)
+	req := httptest.NewRequest("POST", "/api/v1/capturesources", body)
+	req.Header.Set("Content-Type", "application/json")
+
+	app.Test(req, -1)
+
+	// Send request
+	req = httptest.NewRequest("GET", "/api/v1/capturesources?limit=1&offset=0", nil)
+	resp, err := app.Test(req)
+	if err != nil {
+		t.Log(err.Error())
+		t.Fail()
```

Comment:
You can use t.Errorf (if you'd like to mark as fail but continue execution of the test) or t.Fatalf (if you'd like to fail immediately) and then format the error into the failure message. e.g. t.Errorf("did not expect request test to fail: %v",err). Given that you're returning here I suspect you want t.Fatalf.